staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go

```diff
@@ -183,6 +183,23 @@ func WithImpersonation(handler http.Handler, a authorizer.Authorizer, s runtime.
 	})
 }
 
+func userString(u user.Info) string {
+	if u == nil {
```

Comment:
are you sure `u` will be `nil` ? Maybe we could pass the second arg from the `request.UserFrom` ? (line 166).

Reply:
it's never nil in practice. So I think whatever we do here is fine.